Mark query-scheduler ring-based service discovery as stable and document migration (#4028)
grafana/mimir: docs/sources/mimir/operators-guide/deploy-grafana-mimir/jsonnet/migrate-query-scheduler-from-dns-to-ring-based-service-discovery.md

---
description: Learn how to migrate query-scheduler from DNS-based to ring-based service discovery

> menuTitle: Migrate query-scheduler from DNS-based to ring-based service discovery

Reviewer: Would `Migrate query-scheduler` drop too much context from the left-hand TOC?

Author: Unfortunately I think so. We can shorten it to "Migrate query-scheduler to ring-based service discovery"

Author: We agreed on Slack to edit it based on my suggestion.

title: Migrate query-scheduler from DNS-based to ring-based service discovery
weight: 50
---

# Migrate query-scheduler from DNS-based to ring-based service discovery

The query-scheduler supports two service discovery mechanisms:

- [DNS-based service discovery]({{< relref "../../architecture/components/query-scheduler/index.md#dns-based-service-discovery" >}})
- [Ring-based service discovery]({{< relref "../../architecture/components/query-scheduler/index.md#ring-based-service-discovery" >}})

> In this guide you can learn how to migrate query-scheduler from DNS-based to ring-based service discovery when your Mimir cluster is deployed using Jsonnet.

Reviewer: ```suggestion
To migrate the query-scheduler from DNS-based to ring-based service discovery when your Mimir cluster is deployed using Jsonnet:
```

> ## Step 1: Configure the query-scheduler instances to join a ring

Reviewer: ```suggestion
1. Configure the query-scheduler instances to join a ring.
```

> ## Step 1: Configure the query-scheduler instances to join a ring
> 
> Configure the query-scheduler instances to join a ring, but keep querier and query-frontend instances discovering query-schedulers via DNS:

Reviewer: ```suggestion
1. Configure the query-scheduler instances to join a ring, but keep the querier and query-frontend instances discovering query-schedulers via DNS:
```

```jsonnet
{
  _config+:: {
    query_scheduler_service_discovery_mode: 'ring',
    query_scheduler_service_discovery_ring_read_path_enabled: false,
  }
}
```

> ## Step 2: Wait until query-scheduler changes have been applied
> 
> Wait until query-scheduler instances rolled out.
> Then open the [query-scheduler ring status]({{< relref "../../reference-http-api/index.md#query-scheduler-ring-status" >}}) page and ensure all query-scheduler instances are registered to the ring.

Reviewer: ```suggestion
1. Wait until query-scheduler changes have been applied.
1. Open the [query-scheduler ring status]({{< relref "../../reference-http-api/index.md#query-scheduler-ring-status" >}}) page and ensure all query-scheduler instances are registered to the ring.
```

> ## Step 3: Configure query-frontend and querier instances to discover query-schedulers via the ring
> 
> Configure query-frontend and querier instances to discover query-schedulers via the ring:

Reviewer: ```suggestion
1. Configure the query-frontend and querier instances to discover query-schedulers via the ring:
```

Author: I think it would be clear if we keep "Step 3" because it's conceptually different than "Step 2", but let me shorten its title.
